config/src/test/java/org/springframework/security/config/http/OAuth2LoginBeanDefinitionParserTests.java

```diff
@@ -0,0 +1,739 @@
+		assertThat(fieldVal).isInstanceOf(MockClientRegistrationRepository.class);
+		assertThat(fieldVal).isSameAs(clientRegistrationRepository);
+	}
+
+	@Test
+	public void customAuthorizedClientRepository() throws Exception {
+		this.spring.configLocations(this.xml("WithCustomAuthorizedClientRepository")).autowire();
+
+		assertThat(oauth2AuthorizedClientRepository).isInstanceOf(MockOAuth2AuthorizedClientRepository.class);
+
+		Field field = oauth2LoginAuthenticationFilter.getClass().getDeclaredField("authorizedClientRepository");
+		field.setAccessible(true);
+		Object fieldVal = field.get(oauth2LoginAuthenticationFilter);
+		assertThat(fieldVal).isInstanceOf(MockOAuth2AuthorizedClientRepository.class);
+		assertThat(fieldVal).isSameAs(oauth2AuthorizedClientRepository);
+	}
+
+	@Test
+	public void customAuthorizedClientService() throws Exception {
+		this.spring.configLocations(this.xml("WithCustomAuthorizedClientService")).autowire();
+
+		assertThat(oauth2AuthorizedClientService).isInstanceOf(MockOAuth2AuthorizedClientService.class);
+
+		Field authorizedClientRepositoryField = oauth2LoginAuthenticationFilter.getClass()
+				.getDeclaredField("authorizedClientRepository");
+		authorizedClientRepositoryField.setAccessible(true);
+		Object authorizedClientRepositoryFieldVal = authorizedClientRepositoryField
+				.get(oauth2LoginAuthenticationFilter);
+		assertThat(authorizedClientRepositoryFieldVal)
+				.isInstanceOf(AuthenticatedPrincipalOAuth2AuthorizedClientRepository.class);
+
+		Field authorizedClientServiceField = authorizedClientRepositoryFieldVal.getClass()
+				.getDeclaredField("authorizedClientService");
+		authorizedClientServiceField.setAccessible(true);
+		Object authorizedClientServiceFieldVal = authorizedClientServiceField.get(authorizedClientRepositoryFieldVal);
+		assertThat(authorizedClientServiceFieldVal).isInstanceOf(MockOAuth2AuthorizedClientService.class);
+		assertThat(authorizedClientServiceFieldVal).isSameAs(oauth2AuthorizedClientService);
+	}
+
+	@Test
+	public void customAuthorizationRequestRepository() throws Exception {
+		this.spring.configLocations(this.xml("WithCustomAuthorizationRequestRepository")).autowire();
+
+		assertThat(authorizationRequestRepository).isInstanceOf(MockOAuth2AuthorizationRequestRepository.class);
+
+		Field authorizationRequestRepositoryField = oauth2LoginAuthenticationFilter.getClass()
+				.getDeclaredField("authorizationRequestRepository");
+		authorizationRequestRepositoryField.setAccessible(true);
+		Object authorizationRequestRepositoryFieldVal = authorizationRequestRepositoryField
+				.get(oauth2LoginAuthenticationFilter);
+		assertThat(authorizationRequestRepositoryFieldVal).isInstanceOf(MockOAuth2AuthorizationRequestRepository.class);
+		assertThat(authorizationRequestRepositoryFieldVal).isSameAs(authorizationRequestRepository);
+	}
+
+	@Test
+	public void requestWithSingleClientRegistrationAndCustomSuccessAuthenticationHandler() throws Exception {
```

Comment:
Please update test to `verify()` that the mock `AuthenticationSuccessHandler` is called. We can also rename to `requestWhenCustomAuthenticationSuccessHandlerThenCalled`